### ISSUE TITLE
IOS version plans?

### ISSUE THREAD

**Issue by jwknz** [SheaSmith/k-plus#1]
Hi :-)

Do you have any plans for an iOS version of this app?
If not - what are the risks in terms of The API to play around with it? 

Does KAMAR support this API?

**Comment by SheaSmith** [comment]
Currently there are no plans to develop an iOS version of the app (the developer license alone makes the cost prohibitive, plus I don't use iOS devices personally). I think there also might be issues with the App Store review process on iOS, as they're a bit stricter on third party API access than Google Play is.

As for the API, the app primarily uses KAMAR's private API, which is also used by their apps. This means the API is pretty stable and should continue to be supported, barring a major update to their apps. Anecdotally, I've only ever had to deal with one breaking API change in the 3-4 years this app has been using it, and that was a minor issue around what headers are sent up. It also means the API can't really cause any damage (at least for student endpoints, which are basically all read-only, I haven't really looked into the teacher endpoints), as it's limited to what is available to the apps. Theoretically, you could completely imitate the official app in terms of user agent and headers so that KAMAR couldn't tell it was some sort of third party access, but I personally send up unique headers for K+ and am pretty transparent around the fact that it's requests are from a third party.

If you mean legal terms, you'd have to refer to KAMAR's website. My general experience has been that KAMAR's legal team seems reasonably relaxed around third party usage of their API, as long as it is clear that the app is not endorsed by or associated with KAMAR in any way. Saying that, I wouldn't rely on the API for anything critical, as they are probably entitled to ask for any third party apps to be taken offline at any stage (and K+ would comply if requested).

**Comment by jwknz** [comment]
Awesome thanks for the info :-)

I have been playing around with what is possible and seeing what we can be done for something in our school.

We are a Mac based school, hence the iOS question.

Ill keep in touch if I find anything that will be beneficial for you as well. I like the transparency around it - so it would be good to keep that a part of it.

**Comment by SheaSmith** [comment]
No problem.

If you've got any questions around the API, I'd be happy to help. Although most of the bits I've found so far should be reasonably well documented in [this Postman collection](https://documenter.getpostman.com/view/1593669/S1TU1d8Y?version=latest).

**Comment by jwknz** [comment]
Yeah I saw that - which is how I found your repository. Is that document maintained by you as well?

**Comment by SheaSmith** [comment]
Yes. I originally created it for when I was developing K+, but I decided to release it publicly as well.
